making-mobile-services-discoverable-by-metrics.adoc

```diff
@@ -0,0 +1,58 @@
+= Making mobile services auto-discoverable by the metrics service
+
+In order to make a service auto-discoverable by the metrics service:
+
+* Service has to have one of `org.aerogear.metrics/plain_endpoint` or
+`org.aerogear.metrics/ssl_endpoint` annotations.
+
+* Metrics endpoint should have no authentication.
+
+* Metrics endpoint should be accessible by Prometheus instance.
+
+
+Presence of one of the `org.aerogear.metrics/plain_endpoint` or `org.aerogear.metrics/ssl_endpoint`
+annotations makes the service auto-discoverable.
+Value of the annotation will inform metrics service about where to look for metrics data when scraping.
+
+For example, if a service is exposing an HTTP metrics endpoint at
+`/metrics`, the annoation would be:
+
+[source,yaml]
+----
+org.aerogear.metrics/plain_endpoint: "/metrics"
+----
+
+If a service is exposing an HTTPS metrics endpoint at `/rest/metrics`,
+the annoation would be:
+
+[source,yaml]
+----
+org.aerogear.metrics/ssl_endpoint: "/rest/metrics"
+----
+
+This endpoint mustn't have any authentication currently. This is subject to change in the future.
+
+Also, the endpoint should be accessible by the Prometheus instance. There should be a route for the endpoint
+that Prometheus can access. This means, service should be in the same project as Prometheus.
```

Comment:
@david-martin I wrote this sentence according to your comment on the Gist https://gist.github.com/aliok/956d814756bd53d09cf704790f9710e2, but I seconded that the service must be in the same service as Prometheus instance. Is this because of Prometheus' Kubernetes SD limitation?
I am asking because the route doesn't matter here: route could be made accessible by any service on any project.

Reply:
@aliok The endpoint annotation will determine just the endpoint e.g. `/metrics`
The host will be set to the Kbuernetes DNS name of the service, which will only be accessible within the same project as Prometheus, unless project networks are linked.
More info in the docs https://prometheus.io/docs/prometheus/latest/configuration/configuration/#service

Might be good to link to that docs page to say thats what's being used in the prometheus config to discover services (as per @darahayes suggestion)

Reply:
> Kbuernetes DNS name of the service
this is the key: not the public route; K8s DNS name. 
ok, got it now. thanks @david-martin